File: GolandProjects/shapes/rectangle.go
package shapes

type rectangle struct {
	length  int
	breadth int
}

func NewRectangle(length int, breadth int) rectangle {
	return rectangle{length, breadth}
}

func (newRectangle rectangle) Perimeter() int {
	return newRectangle.length*2 + newRectangle.breadth*2
}

File: RubymineProjects/shape/lib/shape/rectangle/rectangle.rb
module Shape

  # Rectangle - represent rectangle with length and breadth
  class Rectangle

    def initialize length:, breadth:
      @length = length
      @breadth = breadth
    end

    def perimeter
      @length * 2 + @breadth * 2
    end

    def area
      if @breadth < 3
        @breadth
      else
        @breadth == @length ? 9 :
          @length == 2 ? 6 :
            (@length == 3 && @breadth == 2) ? 12 : 20
      end

    end
  end
end
